### ISSUE TITLE
`unknown` type isn’t narrowed by 'in' operator

### ISSUE THREAD

**Issue by yaquawa** [microsoft/TypeScript#43774]
### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/C4TwDgpgBAMghiA9gV2FAvFA3lAdnAWwgC4oBnYAJwEtcBzKAXwChmAzZXAY2GsVzyJKBOABtqALwjwkqABSiEKYKU4BrXIgDuuAJSkZy7MyhRqbKAqWooAMltRQkRBcWy06T1ADkiAEYAVhA83nYO3vhEobRQbsq6xqamlBDAyJQCWCZJppEksdbAAHR5ADTZpiyVrKbAABaU2ngQWlAAopSNlMwsQA)

### 💻 Code
```ts
type Layout = { name: string }

function normalizeLayout(layout: unknown): Layout {
  if (layout && typeof layout === 'object' && 'name' in layout) {
    return {
      name: layout.name, // [Error] Property 'name' does not exist on type 'object'.(2339)
    }
  }

  throw new Error
}
```

### 🙁 Actual behavior
Show error even if narrowed by 'in' operator.

### 🙂 Expected behavior
Shouldn't show any error.

**Comment by MartinJohns** [comment]
This is working as intended, and it's unrelated to the type `unknown`. There's simply no way to assume that the type you want is `Layout`, all you know is that you have an `object` that has a `layout` property.

**Comment by yaquawa** [comment]
@MartinJohns I have narrowed down it by `'name' in layout`, but `layout.name` still gives error.

**Comment by MartinJohns** [comment]
See #21732.

**Comment by yaquawa** [comment]
@MartinJohns Thanks!
so currently the `in` operator can't narrowed down properties ? 
Is there any workaround for this?

**Comment by MartinJohns** [comment]
> Is there any workaround for this?

You can use type assertions.

**Comment by yaquawa** [comment]
@MartinJohns OK, thanks!
